### PR TITLE
[Tech 389] Circle refactor

### PR DESCRIPTION
**Pull request by Getriax**
## Changes

Adjusted circle to the ticker requirements:

Available commands:
   * `circle insights`
   * `circle last`
   * `circle clean`

### CODE REVIEW

File: src/circleci/src/circle.messages.js [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    name: 'workflow',
+    style,
+    text,
+    url,
+  };
+};
+
+const insightsMessage = (workflows, username, project, branch) => {
+  const builder = new MessageBuilder({
+    text: '*Circle insights*',
```
Comment (krzyurb) [inline src/circleci/src/circle.messages.js]: Move it to `locales.yml` plz

File: src/circleci/src/circle.messages.js [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+  });
+
+  workflows.forEach(({ name, stats }) => builder.addAttachment({
+    title: name,
+    title_link: `https://circleci.com/gh/${username}/workflows/${project}/tree/${branch}`,
+    text: locales.get('workflow-insights', { ...stats }),
+    footer: locales.get('workflow-insights-footer', { ...stats }),
+    fields: getJobsFileds(stats),
+    actions: [
+      createWorkflowButton('Last succesful', stats.lastSuccessfulWorkflow, 'primary'),
```
Comment (krzyurb) [inline src/circleci/src/circle.messages.js]: Move this and line 43 to locales as well

File: src/circleci/src/circle.messages.js [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    ],
+    color: colors.insights,
+  }));
+
+  return builder.buildMessage();
+};
+
+const lastMessage = (workflows) => {
+  const builder = new MessageBuilder({
+    text: '*Circle last*',
```
Comment (krzyurb) [inline src/circleci/src/circle.messages.js]: Please move it to locales

File: src/circleci/src/circle.messages.js [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+
+  workflows.forEach(({ name, stats }) => builder.addAttachment({
+    title: name,
+    title_link: `https://circleci.com/gh/${username}/workflows/${project}/tree/${branch}`,
+    text: locales.get('workflow-insights', { ...stats }),
+    footer: locales.get('workflow-insights-footer', { ...stats }),
+    fields: getJobsFileds(stats),
+    actions: [
+      createWorkflowButton('Last succesful', stats.lastSuccessfulWorkflow, 'primary'),
+      createWorkflowButton('Last failed', stats.lastFailedWorkflow, 'danger'),
```
Comment (krzyurb) [inline src/circleci/src/circle.messages.js]: Please move it to locales

File: src/circleci/src/extractors/workflows-summary.extractor.js [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+
+const circleSummary = require('../circle.summary');
+const { circleStatsFilter } = require('../filters');
+
+module.exports = (workflows_) => {
+  const workflowsPerName = _.groupBy(workflows_, workflow => workflow.name);
+
+  return _.reduce(Object.entries(workflowsPerName), (workflowSummary, [name, workflows]) => {
+    const workflowsFlatten = _.flatten(workflows);
+    const sortedWorkflows = workflowsFlatten.sort((workflowA, workflowB) => moment(workflowB.startTime).diff(workflowA.startTime, 'ms'));
```
Comment (krzyurb) [inline src/circleci/src/extractors/workflows-summary.extractor.js]: Split that line

File: src/circleci/src/circle.messages.js [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+
+const lastMessage = (workflows) => {
+  const builder = new MessageBuilder({
+    text: '*Circle last*',
+  });
+
+  workflows.forEach(({ name, stats }) => builder.addAttachment({
+    title: name,
+    author_name: stats[stats.lastWorkflowKey].user.login,
+    author_icon: stats[stats.lastWorkflowKey].user.avatar,
```
Comment (krzyurb) [inline src/circleci/src/circle.messages.js]: Is it possible to pass url to user profile?

Comment (Getriax) [reply src/circleci/src/circle.messages.js]: if you mean a GitHub profile like `https://github.com/Getriax` I could create it with `user.login` the property, but there is not a defined one

**Pull request merged**